### PR TITLE
Add stack webhook, project stack gitflow with rbac

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
Address https://github.com/rancher/rio/issues/623

File: tests/integration/rbac_test.go [hunk 1]
```diff
@@ -67,19 +71,48 @@ func rbacTests(t *testing.T, when spec.G, it spec.S) {
...
+			assert.Contains(t, err.Error(), insuffienctPrivilegesMsg)
+			assert.Contains(t, err.Error(), "rio-hostnetwork")
+		})
+
+		it("rio-standard should not be able to create host-port services", func() {
+			testService.Kubeconfig = standardUser.Kubeconfig
+			err := testService.CreateExpectingError(t, "-p", "80,hostport", "nginx")
+			if err == nil {
+				t.Fatal("rio-standard should not be able to create service that enable hostport")
+			}
```
Comment (cbron) [inline tests/integration/rbac_test.go]: You can use `assert.Error` instead of this pattern, below too

File: vendor/modules.txt [hunk 2]
```diff
@@ -51,6 +51,41 @@ github.com/aokoli/goutils
 github.com/armon/go-metrics
 # github.com/avast/retry-go v2.2.0+incompatible
 github.com/avast/retry-go
+# github.com/aws/aws-sdk-go v1.24.1
+github.com/aws/aws-sdk-go/aws
```
Comment (daxmc99) [inline vendor/modules.txt]: Hmm?

File: tests/integration/rbac_test.go (hunk 1 above)
Comment (StrongMonkey) [reply tests/integration/rbac_test.go]: :+1:

File: cli/cmd/up/up.go [hunk 3]
```diff
@@ -191,5 +207,15 @@ func convertObjs(objs []runtime.Object) ([]schema.GroupVersionKind, error) {
 		}
 		gvks = append(gvks, groupVersionKind)
 	}
-	return gvks, nil
+
+	seen := map[string]bool{}
```
[the remark this answers is not in the document]
Comment (StrongMonkey) [reply cli/cmd/up/up.go]: Yeah feel like duplicate gvk doesn't make too much sense. If you have 5 services you will end up having 5 same service gvk while you only need one.

File: vendor/modules.txt (hunk 2 above)
Comment (StrongMonkey) [reply vendor/modules.txt]: go module clean up. maybe QA added in last comment but forgot to run go module

File: pkg/server/startup.go [hunk 4]
```diff
@@ -54,18 +52,18 @@ func Startup(ctx context.Context, systemNamespace, kubeConfig string) error {
 	return nil
 }
 
-func bootstrapResources(rioContext *types.Context, systemNamespace string) (bool, error) {
+func bootstrapResources(rioContext *types.Context, systemNamespace string) error {
 	if _, err := rioContext.Apps.Apps().V1().Deployment().Get(systemNamespace, "rio-controller", metav1.GetOptions{}); errors.IsNotFound(err) {
+		constants.DevMode = true
```
Comment (daxmc99) [inline pkg/server/startup.go]: Hmm, not sure about this. 
If we can't find the controller we just enable dev mode? Seems concerning

File: pkg/webhook/webhook.go [hunk 5]
```diff
@@ -80,6 +80,68 @@ func (w Webhook) Setup() error {
...
+		Groups: request.UserInfo.Groups,
+		Extra:  toExtra(request.UserInfo.Extra),
+	}
+
+	globaleCtx := request2.WithNamespace(request2.WithUser(context.Background(), userInfo), "")
+	if err := rbacregistryvalidation.ConfirmNoEscalation(globaleCtx, h.ruleSolver, globalRules); err != nil {
+		return admissionResponse, fmt.Errorf("failed to validate rio service: %v", err)
+	}
+
+	ctx := request2.WithNamespace(request2.WithUser(context.Background(), userInfo), request.Namespace)
```
Comment (daxmc99) [inline pkg/webhook/webhook.go]: nit: maybe `k8srequest` ?

Comment (StrongMonkey) [reply pkg/webhook/webhook.go]: @daxmc99 addressed

**Pull request merged**